syntax = "proto3";

package frrpc;

enum eCompressType{
	eCompressType_Invilad = 0;
	eCompressType_Not = 1;
}

message RpcRequestMeta{
	int64 request_id = 1;
	uint32 request_size = 2;
}

message RpcResponseMeta{
	repeated int32 sockets = 1;
	uint32 response_size = 2;
}

message RpcMeta{
	string server_name = 1;
	int32 method_index = 2;
	RpcRequestMeta rpc_request_meta = 3;
	RpcResponseMeta rpc_response_meta = 4;
	bytes authentication_data = 5;

	eCompressType compress_type = 10;
}

